server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -402,130 +413,601 @@ describe("Routerlicious", () => {
 								});
-							}),
-							new Promise<void>((resolve) => {
-								secondSocket.on("signal", (signal: ISignalMessage) => {
+							});
+							it("can handle boolean content", async () => {
+								const booleanSignalContent = true;
+								clients[0].socket.send("submitSignal", clients[0].clientId, [
+									booleanSignalContent,
+								]);
+								const userSignals = await Promise.all(
+									clients.map((client) => client.signalPromise),
+								);
+
+								clients.forEach((client, index) => {
 									assert.equal(
-										signalContent,
-										signal.content,
-										"Signal content mismatch",
+										client.signalCount,
+										1,
+										`User ${index + 1} should have received 1 signal`,
 									);
 									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
+										userSignals[index].content,
+										booleanSignalContent,
+										`User ${index + 1} signal content mismatch`,
 									);
-									resolve();
 								});
-							}),
-						];
+							});
+							it("can handle JSON/Object content", async () => {
+								const objectSignalContent = {
+									key1: "value1",
+									key2: 42,
+									key3: true,
+								};
+								clients[0].socket.send("submitSignal", clients[0].clientId, [
+									objectSignalContent,
+								]);
+								const userSignals = await Promise.all(
+									clients.map((client) => client.signalPromise),
+								);
 
-						firstSocket.send("submitSignal", firstConnectMessage.clientId, [
-							signalContent,
-						]);
-						await Promise.all(promises);
-					});
+								clients.forEach((client, index) => {
+									assert.equal(
+										client.signalCount,
+										1,
+										`User ${index + 1} should have received 1 signal`,
+									);
+									assert.deepEqual(
+										userSignals[index].content,
+										objectSignalContent,
+										`User ${index + 1} signal content mismatch`,
+									);
+								});
+							});
+							it("can handle malformed JSON content", async () => {
+								const malformedJsonSignalContent =
+									"{key1: value1, key2: 42, key3: true}";
+								clients[0].socket.send("submitSignal", clients[0].clientId, [
+									malformedJsonSignalContent,
+								]);
+								const userSignals = await Promise.all(
+									clients.map((client) => client.signalPromise),
+								);
 
-					it("Multiple Signals", async () => {
-						const signalContent1 = "TestSignal1";
-						const signalContent2 = "TestSignal2";
+								clients.forEach((client, index) => {
+									assert.equal(
+										client.signalCount,
+										1,
+										`User ${index + 1} should have received 1 signal`,
+									);
+									assert.deepEqual(
+										userSignals[index].content,
+										malformedJsonSignalContent,
+										`User ${index + 1} signal content mismatch`,
+									);
+								});
+							});
+						});
+					});
```

Comment:
There is other information returned besides content. Can we validate expectations there?

Reply:
Whoops I thought you were referring to ref seq # and client connection number. Will add.